src/layout/layout.rs

```diff
@@ -479,383 +539,379 @@ impl Layout {
+        let spacer_rects = changes_to_rects(&changes, &spacers, inner_area, self.direction);
+
+        Ok((segment_rects, spacer_rects))
+    }
+}
+
+fn configure_area(
+    solver: &mut Solver,
+    area: Element,
+    area_start: f64,
+    area_end: f64,
+) -> Result<(), AddConstraintError> {
+    solver.add_constraint(area.start | EQ(REQUIRED) | area_start)?;
+    solver.add_constraint(area.end | EQ(REQUIRED) | area_end)?;
+    Ok(())
+}
 
-        match flex {
-            Flex::SpaceBetween => {
-                let spacers: Vec<Element> = std::iter::repeat_with(|| {
-                    Element::constrain(&mut solver, (area_start, area_end))
-                })
-                .take(elements.len().saturating_sub(1)) // one less than the number of elements
-                .try_collect()?;
-                // spacers growing should be the lowest priority
-                for spacer in spacers.iter() {
-                    solver.add_constraint(spacer.size() | EQ(WEAK) | area_size)?;
-                }
-                // Spacers should all be similar in size
-                // these constraints should not be stronger than existing constraints
-                // but if they are weaker `Min` and `Max` won't be pushed to their desired values
-                // I found using `STRONG` gives the most desirable behavior
-                for (left, right) in spacers.iter().tuple_combinations() {
-                    solver.add_constraint(left.size() | EQ(STRONG) | right.size())?;
-                }
-                // interleave elements and spacers
-                // for `SpaceBetween` we want the following
-                // `[element, spacer, element, spacer, ..., element]`
-                // this is why we use one less spacer than elements
-                for pair in Itertools::interleave(elements.iter(), spacers.iter())
-                    .collect::<Vec<&Element>>()
-                    .windows(2)
-                {
-                    solver.add_constraint(pair[0].end | EQ(REQUIRED) | pair[1].start)?;
-                }
+fn configure_variable_constraints(
+    solver: &mut Solver,
+    variables: &[Variable],
+    area: Element,
+) -> Result<(), AddConstraintError> {
+    // all variables are in the range [area.start, area.end]
+    for &variable in variables.iter() {
+        solver.add_constraint(variable | GE(REQUIRED) | area.start)?;
+        solver.add_constraint(variable | LE(REQUIRED) | area.end)?;
+    }
+
+    // all variables are in ascending order
+    for (&left, &right) in variables.iter().tuple_windows() {
+        solver.add_constraint(left | LE(REQUIRED) | right)?;
+    }
+
+    // the first and last variables are at the start and end of the area
+    if let (Some(&first), Some(&last)) = (variables.first(), variables.last()) {
+        solver.add_constraint(first | EQ(REQUIRED) | area.start)?;
+        solver.add_constraint(last | EQ(REQUIRED) | area.end)?;
+    }
```

Comment:
We could also move this to the main split method as something like
```rust
let area = Element::from((first.start, last.end))
```

Reply:
I did this:

```rust
let area_size = Element::from((*variables.first().unwrap(), *variables.last().unwrap()));
```

using `.unwrap()` since they always exist. `variables[0]` and `variables[variables.len()-1]` also works for me. Is there one you prefer?

Reply:
I'm fine with either.